controllers/managedocs_controller.go

```diff
@@ -1598,6 +1599,27 @@ func (r *ManagedOCSReconciler) reconcileEgressNetworkPolicy() error {
 			)
 		}
 
+		if r.DeploymentType == consumerDeploymentType {
+
+			storageProviderEndpointURL := r.addonParams[storageProviderEndpointKey]
+			storageProviderEndpointHost := strings.Split(storageProviderEndpointURL, ":")[0]
```

Comment:
This might panic if the split is yielding an empty array in the case of a malformed URL.

Reply:
Added a condition to return a error if the split is yielding an empty string